Core/PrivatTransactionsImporter.cs

```diff
@@ -9,72 +8,84 @@
-            {
-                handler(await LoadData(date, credentials));
-            }
-        }
+	public class PrivatTransactionsImporter
+	{
+		public static async Task ImportTransactions(Credentials credentials, Action<List<Transaction>> handler)
+		{
+			for (var date = DateTime.Now.ToDate(); date.Year > 2015; date -= TimeSpan.FromDays(1))
+			{
+				handler(await LoadData(date, credentials));
+			}
+		}
 
-        private static async Task<List<Transaction>> LoadData(Date date, Credentials credentials)
-        {           
-            var data = $"<oper>cmt</oper><wait>60</wait><payment><prop name=\"sd\" value=\"{date.Day}.{date.Month}.{date.Year}\"/><prop name=\"ed\" value=\"{date.Day}.{date.Month}.{date.Year}\"/><prop name=\"card\" value=\"{credentials.CardNumber}\"/></payment>";
-            var md5 = MD5.Create();
-            var sha1 = SHA1.Create();
+		private static async Task<List<Transaction>> LoadData(Date date, Credentials credentials)
+		{
+			var data = $"<oper>cmt</oper><wait>60</wait><payment><prop name=\"sd\" value=\"{date.Day}.{date.Month}.{date.Year}\"/><prop name=\"ed\" value=\"{date.Day}.{date.Month}.{date.Year}\"/><prop name=\"card\" value=\"{credentials.CardNumber}\"/></payment>";
+			var md5 = MD5.Create();
+			var sha1 = SHA1.Create();
 
-            var secret = data + credentials.MerchantPassword;
-            var bytes = Encoding.UTF8.GetBytes(secret);
-            var md5h = md5.ComputeHash(bytes);
-            var md5hs = ByteArrayToString(md5h);
-            var sha1h = sha1.ComputeHash(Encoding.UTF8.GetBytes(md5hs));
-            var signature = ByteArrayToString(sha1h);
+			var secret = data + credentials.MerchantPassword;
+			var bytes = Encoding.UTF8.GetBytes(secret);
+			var md5h = md5.ComputeHash(bytes);
+			var md5hs = ByteArrayToString(md5h);
+			var sha1h = sha1.ComputeHash(Encoding.UTF8.GetBytes(md5hs));
+			var signature = ByteArrayToString(sha1h);
 
-            HttpClient client = new HttpClient();
-            var request = new HttpRequestMessage
-            {
-                Method = HttpMethod.Get,
-                RequestUri = new Uri("https://api.privatbank.ua/p24api/rest_fiz"),
-                Content = new StringContent($"<?xml version=\"1.0\" encoding=\"UTF-8\"?><request version=\"1.0\"><merchant><id>{credentials.MerchantId}</id><signature>{signature}</signature></merchant><data>{data}</data></request>",
-                    Encoding.Default, "application/xml"),
-            };
-            var res = await client.SendAsync(request);
+			var client = new HttpClient();
+			var request = new HttpRequestMessage
+			{
+				Method = HttpMethod.Get,
+				RequestUri = new Uri("https://api.privatbank.ua/p24api/rest_fiz"),
+				Content = new StringContent($"<?xml version=\"1.0\" encoding=\"UTF-8\"?><request version=\"1.0\"><merchant><id>{credentials.MerchantId}</id><signature>{signature}</signature></merchant><data>{data}</data></request>", Encoding.Default, "application/xml")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Split this 257 characters long line (which is greater than 200 authorized).](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=4931949)